be/src/service/doris_main.cpp

```diff
@@ -73,17 +75,25 @@ static void thrift_output(const char* x) {
 
 }
 
+DECLARE_bool(help);
+DECLARE_bool(version);
+DEFINE_string(tool, "", "tools for BE, including: meta, segment");
+DEFINE_bool(h, false, "equal to --help, print help info");
+
 int main(int argc, char** argv) {
     // check if print version or help
-    if (argc > 1) {
-        if (strcmp(argv[1], "--version") == 0 || strcmp(argv[1], "-v") == 0) {
-            puts(doris::get_build_version(false).c_str());
-            exit(0);
-        } else if (strcmp(argv[1], "--help") == 0 || strcmp(argv[1], "-?") == 0) {
-            help(basename(argv[0]));
-            exit(0);
-        }
+    google::ParseCommandLineNonHelpFlags(&argc, &argv, true);
+    if (FLAGS_help || FLAGS_h) {
+        help(basename(argv[0]));
+        exit(0);
+    } else if (FLAGS_version || FLAGS_v) {
+        std::cout << doris::get_build_version(false) << std::endl;
+        exit(0);
+    } else if (FLAGS_tool != "") {
+        doris::Tools::process(FLAGS_tool);
```

Comment:
Emmm, I think it is not a good idea to use our doris_be as a debug tool. It will be better to seperate tool and core. Because this will make two independent things together, it will make a simple thing more complicated. If tool has been implemented in this way, when admin want to update doris tool, he must update the doris_be binary, it is not efficient.